server/modules/suricata/suricata.go

```diff
@@ -283,6 +396,140 @@ func (e *SuricataEngine) IsRunning() bool {
 	return e.isRunning
 }
 
+// Rule file operations
+
+// SyncLocalDetections - DEPRECATED: Use RegenerateRuleFiles instead
+// This is called when a rule is changed (eg. via the UI)
+func (e *SuricataEngine) SyncLocalDetections(ctx context.Context, detections []*model.Detection) (errMap map[string]string, err error) {
+	e.logger().
+		Debug("SyncLocalDetections called - delegating to RegenerateRuleFiles")
+	return e.RegenerateRuleFiles(ctx, detections)
+}
+
+// RegenerateRuleFiles regenerates all rule files from current Elasticsearch state.
+func (e *SuricataEngine) RegenerateRuleFiles(ctx context.Context, changedDetections []*model.Detection) (errMap map[string]string, err error) {
+	logger := e.logger().WithFields(log.Fields{
+		"changedCount": len(changedDetections),
+		"operation":    "regenerateRuleFiles",
+	})
+
+	// Check for sync block
+	if blocked := detections.IsSyncBlocked(e.IOManager, e.SyncBlockFilePath); blocked {
+		logger.WithFields(log.Fields{
+			"blockFilePath": e.SyncBlockFilePath,
+		}).Warn("regenerate rule files blocked by block file")
+
+		return map[string]string{"blocked": "sync operations blocked"},
+			fmt.Errorf("rule file regeneration blocked")
+	}
+
+	// Log what triggered this regeneration (if provided)
+	if len(changedDetections) > 0 {
+		sampleSIDs := []string{}
+		for i, det := range changedDetections {
+			if i < 5 { // Log first 5 SIDs as sample
+				sampleSIDs = append(sampleSIDs, det.PublicID)
+			} else {
+				break
+			}
+		}
+		logger.WithField("sampleSIDs", sampleSIDs).Info("regenerating rule files after detection changes")
+	} else {
+		logger.Info("regenerating rule files")
+	}
+
+	// Get ALL detections from Elasticsearch (source of truth) - needed for flowbits handling
+	allDetectionsMap, err := e.getAllSuricataDetections(ctx)
+	if err != nil {
+		logger.WithError(err).Error("failed to fetch detections from store")
+		return map[string]string{"fetch": err.Error()}, err
+	}
+
+	// Filter out pending delete
+	allDetections := make([]*model.Detection, 0, len(allDetectionsMap))
+	for _, det := range allDetectionsMap {
+		if !det.PendingDelete {
+			allDetections = append(allDetections, det)
+		}
+	}
+
+	logger.WithField("totalDetections", len(allDetections)).Debug("fetched detections")
+
+	// Apply filters to maintain consistency with sync behavior
+	for _, det := range allDetections {
+		e.ApplyFilters(det)
```

Comment:
Doublecheck this

Reply:
Removed. ES already has filtered state from last sync, and regex config changes trigger a force sync on restart.